ipam/allocator.go

```diff
@@ -680,13 +690,15 @@ func (alloc *Allocator) reportFreeSpace() {
 // Owned addresses
 
 func (alloc *Allocator) addOwned(ident string, addr address.Address) {
-	alloc.owned[ident] = addr
+	alloc.owned[ident] = append(alloc.owned[ident], addr)
```

Comment:
We are assuming that `addr` isn't already in the list here, otherwise we end up with duplicate entries which `Free` won't remove.

This is guaranteed by the calling context, but it took me a while to figure out. Not sure what to do about that; add a note perhaps, "NB: addr must not be owned by ident already"?